tee-worker/app-libs/stf/src/trusted_call_litentry.rs

```diff
@@ -274,45 +179,62 @@ impl TrustedCallSigned {
 		})
 	}
 
-	pub fn set_challenge_code_runtime(
-		enclave_account: AccountId,
+	pub fn verify_identity_callback_internal(
+		signer: AccountId,
 		who: AccountId,
 		identity: Identity,
-		code: ChallengeCode,
-	) -> StfResult<()> {
-		ensure_enclave_signer_account(&enclave_account)?;
-		// only used in tests, we don't care about the error
-		IMTCall::set_challenge_code { who, identity, code }
-			.dispatch_bypass_filter(RuntimeOrigin::root())
-			.map_err(|e| StfError::Dispatch(format!("{:?}", e.error)))?;
-		Ok(())
-	}
-
-	// internal helper fn
-	fn create_identity_runtime_internal(
-		who: AccountId,
-		identity: Identity,
-		metadata: Option<MetadataOf<Runtime>>,
 		bn: ParentchainBlockNumber,
-		parent_ss58_prefix: u16,
-	) -> StfResult<ChallengeCode> {
-		IMTCall::create_identity {
+	) -> StfResult<UserShieldingKeyType> {
+		// important! The signer has to be enclave_signer_account, as this TrustedCall can only be constructed internally
+		ensure_enclave_signer_account(&signer)
+			.map_err(|_| StfError::VerifyIdentityFailed(ErrorDetail::UnauthorisedSender))?;
+
+		let key = IdentityManagement::user_shielding_keys(&who)
+			.ok_or(StfError::VerifyIdentityFailed(ErrorDetail::UserShieldingKeyNotFound))?;
+
+		IMTCall::verify_identity {
 			who: who.clone(),
 			identity: identity.clone(),
-			metadata,
-			creation_request_block: bn,
-			parent_ss58_prefix,
+			verification_request_block: bn,
 		}
 		.dispatch_bypass_filter(RuntimeOrigin::root())
-		.map_err(|e| StfError::CreateIdentityFailed(e.into()))?;
+		.map_err(|e| StfError::VerifyIdentityFailed(e.into()))?;
 
-		// generate challenge code
-		let code = generate_challenge_code();
+		// remove challenge code
+		IMTCall::remove_challenge_code { who, identity }
```

Comment:
Not now, but might be in the future. That's why we have a dedicated sidechain extrinsic for it (meaning the user might be able to clear it explicitly).

Besides that, there's a `set_challenge_code` fn, it would be strange if there's no counterpart of it.

Hence we will probably leave it unchanged.

Reply:
We can have `remove_challenge_code` as well and same logic inside `verify_identity` if these functions should always be called together in case of veryfing identity

Reply:
That's indeed better for the current situation, thanks.

One slight concern is what if `remove_challenge_code` somehow fails, should the identity still considered verified? (My preference is yes). In that case it might be better to separate those two, and we shouldn't use `?` on `remove_challenge_code` ofc

Reply:
@Kailai-Wang sorry for late reply.
From the user's perspective, your preference sounds like the correct way of handling such errors. We could monitor dangling challenge codes and think later about what to do with them.